### PR TITLE
DBML Language support

### PR DESCRIPTION
**Pull request by ghashy**
It would be great if we can read/write DBML files in Helix.
[DBML](https://dbml.dbdiagram.io/home/) (Database Markup Language) is an open-source language for documenting database schemas and structures.
The highlighting is not complete, but it is good starting point:
Screenshot 2023-11-19 at 13 25 24

### CODE REVIEW

File: languages.toml [hunk 1]
```diff
@@ -2928,3 +2928,16 @@ language-servers = [ "templ" ]
 [[grammar]]
 name = "templ"
 source = { git = "https://github.com/vrischmann/tree-sitter-templ", rev = "ea56ac0655243490a4929a988f4eaa91dfccc995" }
+
+[[language]]
+name = "dbml"
+scope = "source.dbml"
+injection-regex = "dbml"
+file-types = ["dbml"]
+roots = []
```
Comment (the-mikedavis) [inline languages.toml:2937]: This `roots` key can be omitted if you merge/rebase on the latest master

File: runtime/queries/dbml/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+("TableGroup" ) @type
+("database_type" ) @variable
+
+; Index related highlighting
+(index_block) @class
+
+; string and number constants
+("'''") @string.escape
+(string) @string
+(number) @constant
```
Comment (the-mikedavis) [inline runtime/queries/dbml/highlights.scm:30]:
```suggestion
(number) @constant.numeric
```

File: runtime/queries/dbml/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+("Project" ) @type
+("Table" ) @type
+("TableGroup" ) @type
+("database_type" ) @variable
+
+; Index related highlighting
+(index_block) @class
+
+; string and number constants
+("'''") @string.escape
```
Comment (the-mikedavis) [inline runtime/queries/dbml/highlights.scm:28]:
```suggestion
("'''") @constant.character.escape
```

File: runtime/queries/dbml/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  "["
+  "]"
+] @punctuation.bracket
+
+; brackets
+[
+  ":"
+  "."
+  ","
+] @punctuation.delimeter
```
Comment (the-mikedavis) [inline runtime/queries/dbml/highlights.scm:47]:
```suggestion
] @punctuation.delimiter
```

File: runtime/queries/dbml/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+(type) @keyword
+
+; Highlight special types for database/data types
+("Project" ) @type
+("Table" ) @type
+("TableGroup" ) @type
+("database_type" ) @variable
+
+; Index related highlighting
+(index_block) @class
```
Comment (the-mikedavis) [inline runtime/queries/dbml/highlights.scm:25]: `@class` isn't a capture name that we use, see https://docs.helix-editor.com/master/themes.html#syntax-highlighting

File: languages.toml (hunk 1 above)
Comment (ghashy) [reply languages.toml]: line is removed :)

File: runtime/queries/dbml/highlights.scm (hunk 5 above)
Comment (ghashy) [reply runtime/queries/dbml/highlights.scm]: I've noticed that the 'block' seems to have no noticeable effect on the syntax highlighting, removed it

File: runtime/queries/dbml/highlights.scm [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
```
Comment (the-mikedavis) [inline runtime/queries/dbml/highlights.scm:24]: Same here about `@class`

Comment (ghashy) [reply runtime/queries/dbml/highlights.scm]: removed it

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**